lib/awful/Ezkey.lua

```diff
@@ -0,0 +1,129 @@
+---------------------------------------------------------------------------
+--- Define easily new key and button objects.
+--
+-- Built on top awful.key and awful.button and also aims
+-- at decluttering your config.
+--
+-- Use awful.EzKey to define a keybinding
+--
+-- This example shows how to define a basic EzKey object:
+--
+-- awful.EzKey("C-M-s",function() awful.util.spawn("brave") end,"some description","some group name")
+--
+-- This example shows how to define a basic EzButton object:
+--
+-- awful.EzButton("M-1",function() .... end)
+--
+-- @author neon_arch neon_arch@gitlab.com;
+-- @copyright 2022 neon_arch
+-- @inputmodule awful.Ezkey
+---------------------------------------------------------------------------
+
+
+-------------imports---------------------
+local key = require("key");
+local easykey = {};
+
+---------------------Ezkey---------------------
+
+
+--#region
+-- A function to handle spliting of key group like "C-M-s" into
+-- a table of keys something like this {"C","M","s"}
+-- #endregion
+local function split(inputstr, sep)
+    if sep == nil then
+        sep = "%s"
+    end
+    local split_string = {}
+    for str in string.gmatch(inputstr, "([^" .. sep .. "]+)") do
+        table.insert(split_string, str)
+    end
+    return split_string
+end
+
+--#region
+--function to check whether the last key is a number or not
+--if it is then convert it into a number and return it
+--#endregion
+local function check_if_number(key)
+    if (key == "1") then
```

Comment:
you can use `tonunmber` for this

Reply:
I will improve it in the next PR, no problem.